### ISSUE TITLE
Error [pose_visualizer.py, 95] Heatmap size is not valid 

### ISSUE THREAD

**Issue by eunjoa** [youansheng/pytorch-cv#4]
Hello, i've run the train code for pose without modifying much in the code (beside paths), but the call of vis_paf rises the error 'heat map size is not valid' . printing the inputs size returns: [1, 38, 46, 46] which is 4 dims which seems to be right if the first dim is for batch (here 1), why the input size lenght is tested to be 3 dims here. am i missing on something ? thanks
  
` def vis_paf(self, inputs, ori_img, name='default', vis_dir=PAF_DIR):
        vis_dir = os.path.join(self.configer.get('project_dir'), vis_dir)
        if not os.path.exists(vis_dir):
            Log.error('Dir:{} not exists!'.format(vis_dir))
            os.makedirs(vis_dir)

        if not isinstance(inputs, np.ndarray):
            print("inputs size : {}".format(inputs.size()))
            **if len(inputs.size()) != 3:
                Log.error('Heatmap size is not valid.')
                exit(1)**

            inputs = inputs.data.squeeze().cpu().numpy().transpose(1, 2, 0)

        if not isinstance(ori_img, np.ndarray):
            if len(ori_img.size()) != 3:
                Log.error('Heatmap size is not valid.')
                exit(1)

            ori_img = DeNormalize(mean=[128.0, 128.0, 128.0],std=[256.0, 256.0, 256.0])(ori_img)
            ori_img = ori_img.data.cpu().squeeze().numpy().transpose(1, 2, 0).astype(np.uint8)

        if inputs.shape[0] != ori_img.shape[0]:
            resize_width = (inputs.shape[1] + ori_img.shape[1]) // 2
            resize_height = (inputs.shape[0] + ori_img.shape[0]) // 2
            inputs = cv2.resize(inputs, (resize_width, resize_height), interpolation=cv2.INTER_CUBIC)
            ori_img = cv2.resize(ori_img, (resize_width, resize_height), interpolation=cv2.INTER_CUBIC)

        for i in range(1):
            U = inputs[:, :, 2*i] * -1
            V = inputs[:, :, 2*i+1]
            X, Y = np.meshgrid(np.arange(U.shape[1]), np.arange(U.shape[0]))
            M = np.zeros(U.shape, dtype='bool')
            M[U ** 2 + V ** 2 < 0.5 * 0.5] = True
            U = ma.masked_array(U, mask=M)
            V = ma.masked_array(V, mask=M)

            # 1
            plt.figure()
            plt.imshow(ori_img+0.5, alpha=.5)
            s = 5
            Q = plt.quiver(X[::s, ::s], Y[::s, ::s], U[::s, ::s], V[::s, ::s],
                           scale=50, headaxislength=4, alpha=.5, width=0.001, color='r')

            fig = matplotlib.pyplot.gcf()
            fig.set_size_inches(10, 10)
            plt.savefig(os.path.join(vis_dir, '{}_{}.jpg'.format(name, i)))
`

**Comment by youansheng** [comment]
I will update it later.

**Comment by aigeek** [comment]
I have the same problem:
Error [pose_visualizer.py, 93] inputs.size=torch.Size([1,38,46,46])
Error [pose_visualizer.py, 95] Heatmap size is not valid

I changed nothing just to run the coco code:
python main.py --hypes hypes/pose/coco/op_coco_pose.json \
               --base_lr 0.001 \
               --phase train
What can I do to fix the bug?

**Comment by eunjoa** [comment]
@aigeek you can fix that by squeezing inputs before passing them to the the vis function, or just change the condition to` if len(inputs.size()) != 4`, and if you don't need visualization during your training you can just comment the visualization part.

**Comment by lirus7** [comment]
So finally how to fix this error? 
When I removed the visualization part I got a sizes do not match error?
What should be done for this?
